File: SpriteKitLib/src/main/java/br/com/insanitech/spritekit/opengl/model/GLRectangle.java
package br.com.insanitech.spritekit.opengl.model;

/**
 * Created by anderson on 7/3/15.
 */
public class GLRectangle extends GLGeometry {
    public GLRectangle() {
        vertices = new float[] {
                0.0f, 0.0f, 0.0f,
                1.0f, 0.0f, 0.0f,
                0.0f, 1.0f, 0.0f,
                1.0f, 1.0f, 0.0f
        };
        componentsPerVertice = 2;
        generateVertex();
    }
}

File: SpriteKitLib/src/main/java/br/com/insanitech/spritekit/opengl/model/GLRectangle.kt
package br.com.insanitech.spritekit.opengl.model

/**
 * Created by anderson on 7/3/15.
 */
internal class GLRectangle : GLGeometry() {
    init {
        this.vertices = floatArrayOf(0.0f, 0.0f, 0.0f, 1.0f, 0.0f, 0.0f, 0.0f, 1.0f, 0.0f, 1.0f, 1.0f, 0.0f)
        this.componentsPerVertices = 2
        generateVertex()
    }
}
